components/script/body.rs

```diff
@@ -27,21 +32,24 @@ pub enum BodyType {
     Blob,
     FormData,
     Json,
-    Text
+    Text,
+    ArrayBuffer
 }
 
 pub enum FetchedData {
     Text(String),
     Json(JSValue),
     BlobData(DomRoot<Blob>),
     FormData(DomRoot<FormData>),
+    ArrayBuffer(Heap<JSVal>),
```

Comment:
To ensure this code is GC-safe, we should use `RootedTraceableBox<Heap<JSVal>>`, and use it for the Json variant as well.